python-mini-projects/projects/Web_page_summation/utils/train.py

```diff
@@ -0,0 +1,106 @@
+import os
+import tensorflow as tf
+import time
+from model import Model
+from utils import build_dict, build_dataset, batch_iter
+start = time.perf_counter()
+default_path = '.'
+
+
+class args:
+    pass
+
+
+args.num_hidden = 150
+args.num_layers = 2
+args.beam_width = 10
+args.glove = "store_true"
+args.embedding_size = 300
+
+args.learning_rate = 1e-3
+args.batch_size = 64
+args.num_epochs = 10
+args.keep_prob = 0.8
+
+args.toy = False  # "store_true"
+
+args.with_model = "store_true"
+
+
+if not os.path.exists(default_path + "saved_model"):
+    os.mkdir(default_path + "saved_model")
+else:
+    # if args.with_model:
+    old_model_checkpoint_path = open(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *old\_model\_checkpoint\_path*. The resource is allocated by call *builtins.open*. Execution paths that do not contain closure statements were detected. To prevent this resource leak, close *old\_model\_checkpoint\_path* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).